### PR TITLE
Transformer code for DisplayNames component.

### PR DESCRIPTION
**Pull request by snktd**
This is a replica for https://github.com/unicode-org/icu4x/pull/2592. I had tons of merge conflicts there and somehow messed that PR up.

### CODE REVIEW

File: components/icu/Cargo.toml [hunk 1]
```diff
@@ -55,6 +56,7 @@ std = [
     "icu_collections/std",
     "icu_datetime/std",
     "icu_decimal/std",
+    # "icu_displaynames/std",
```
Comment (robertbastian) [inline components/icu/Cargo.toml:59]: We can use `icu_displaynames?/std` to not accidentally activate the dependency. Please update the other commented-out deps to that as well.

File: experimental/displaynames/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! [`icu_displaynames`](crate) is one of the ['ICU4X`] components.
+//!
+//! TThis API provides a mapping of locale display names for regions.
```
Comment (robertbastian) [inline experimental/displaynames/src/lib.rs:7]: Typo

File: experimental/displaynames/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+//!
+//! TODO: expand documentation
```
Comment (robertbastian) [inline experimental/displaynames/src/lib.rs:9]: Please add at least the boilerplate that this is experimental and that this is both a module in `icu` as well as its own crate. See `icu_segmenter`.

File: provider/datagen/src/transform/cldr/displaynames/mod.rs [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+const ALT_VARIANT_SUFFIX: &str = "-alt-variant";
+
+impl From<&cldr_serde::displaynames::Resource> for TerritoryDisplayNamesV1<'static> {
+    fn from(other: &cldr_serde::displaynames::Resource) -> Self {
+        let mut names = ZeroMap::new();
+        for lang_data_entry in other.main.0.iter() {
+            for entry in lang_data_entry.1.localedisplaynames.territories.iter() {
+                let region = entry.0;
+                if !region.ends_with(ALT_VARIANT_SUFFIX) {
+                    if let Ok(key) = <TinyAsciiStr<3>>::from_str(region) {
```
Comment (robertbastian) [inline provider/datagen/src/transform/cldr/displaynames/mod.rs:56]: Should this error be handled?

File: components/icu/Cargo.toml (hunk 1 above)
Comment (snktd) [reply components/icu/Cargo.toml]: Done.

File: experimental/displaynames/src/lib.rs (hunk 3 above)
Comment (snktd) [reply experimental/displaynames/src/lib.rs:15]: Done! After this change, even if I run 'cargo make generate-readmes' first, I am getting an error that read-me file is out of sync with lib.rs.

File: provider/datagen/src/transform/cldr/displaynames/mod.rs (hunk 4 above)
Comment (snktd) [reply provider/datagen/src/transform/cldr/displaynames/mod.rs]: Thanks for this comment. I figured that there are some regions with "alt-short" suffix which weren't removed and the error wasn't bubbling up. Added the changes to address this.

File: experimental/displaynames/src/lib.rs (hunk 2 above)
Comment (snktd) [reply experimental/displaynames/src/lib.rs]: Done.

File: experimental/displaynames/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,21 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! [`icu_displaynames`](crate) is one of the ['ICU4X`] components.
+//!
+//! This API provides a mapping of locale display names for regions.
```
Comment (robertbastian) [inline experimental/displaynames/src/lib.rs:7]:
```suggestion
//! 🚧 \[Experimental\] Display names for regions
//!
//! This module is published as its own crate ([`icu_displaynames`](https://docs.rs/icu_displaynames/latest/icu_displaynames/))
//! and as part of the [`icu`](https://docs.rs/icu/latest/icu/) crate. See the latter for more details on the ICU4X project.
```

File: experimental/displaynames/src/lib.rs [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+//!
+//! <div class="stab unstable">
+//! 🚧 This code is experimental; it may change at any time, in breaking or non-breaking ways,
+//! including in SemVer minor releases. It can be enabled with the "experimental" feature
+//! of the icu meta-crate. Use with caution.
+//! </div>
+//!
+//! TODO: expand documentation
```
Comment (robertbastian) [inline experimental/displaynames/src/lib.rs:15]: take this out of the doc comment into a normal comment

File: experimental/displaynames/Cargo.toml [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+tinystr = { version = "0.7", path = "../../utils/tinystr", features = ["alloc", "zerovec"], default-features = false }
+
+[lib]
+path = "src/lib.rs"
+bench = false  # This option is required for Benchmark CI
+
+[features]
+std = ["icu_provider/std"]
+default = []
+serde = ["dep:serde", "zerovec/serde", "icu_collections/serde"]
```
Comment (robertbastian) [inline experimental/displaynames/Cargo.toml:42]:
```suggestion
serde = ["dep:serde", "zerovec/serde", "icu_collections/serde", "tinystr/serde"]
```

File: experimental/displaynames/Cargo.toml [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+    "Cargo.toml",
+    "LICENSE",
+    "README.md"
+]
+
+[dependencies]
+icu_collections = { version = "1.0.0", path = "../../components/collections" }
+icu_locid = { version = "1.0.0", path = "../../components/locid" }
+icu_provider = { version = "1.0.0", path = "../../provider/core", features = ["macros"] }
+yoke = { version = "0.6.0", path = "../../utils/yoke", features = ["derive"] }
```
Comment (robertbastian) [inline experimental/displaynames/Cargo.toml:29]: This shouldn't be needed as it's reeported in `icu_provider::prelude::*`
```suggestion
```

**Review by robertbastian — COMMENTED** [review]
You need to run `cargo make diplomat-coverage`

File: experimental/displaynames/src/lib.rs (hunk 6 above)
Comment (snktd) [reply experimental/displaynames/src/lib.rs]: Done.

**Comment by snktd** [comment]
> 

Exec cargo make diplomat-coverage.

**Review by robertbastian — APPROVED** [review]

**Review by sffc — COMMENTED** [review]
LGTM

**Pull request merged**